packages/pymedphys_deliverydata/src/pymedphys_deliverydata/dicom/to_dicom.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (C) 2019 Cancer Care Associates and Simon Biggs
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version (the "AGPL-3.0+").
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License and the additional terms for more
+# details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+# ADDITIONAL TERMS are also included as allowed by Section 7 of the GNU
+# Affero General Public License. These additional terms are Sections 1, 5,
+# 6, 7, 8, and 9 from the Apache License, Version 2.0 (the "Apache-2.0")
+# where all references to the definition "License" are instead defined to
+# mean the AGPL-3.0+.
+
+# You should have received a copy of the Apache-2.0 along with this
+# program. If not, see <http://www.apache.org/licenses/LICENSE-2.0>.
+
+
+from copy import deepcopy
+
+import numpy as np
+
+from pymedphys_base.deliverydata import DeliveryDataBase
+
+from pymedphys_dicom.rtplan import (
+    get_gantry_angles_from_dicom,
+    build_control_points,
+    replace_fraction_group,
+    replace_beam_sequence,
+    restore_trailing_zeros,
+    merge_beam_sequences,
+    get_fraction_group_index,
+    convert_to_one_fraction_group)
+
+from ..utilities import get_all_masked_delivery_data
+
+
+def delivery_data_to_dicom(filtered_delivery_data: DeliveryDataBase,
```

Comment:
Why does this need to be a separate function? I really think it's overgeneralising at the expense of readability. I think it is vanishingly unlikely that this function will (or at least *should*) ever be used on a `DeliveryDataBase`-like object...

Reply:
Fair point... can we leave it like this for a bit? And let it settle. I think it will be worth revisiting delivery data after the DICOM objects are built...